routes/baseDBMethods.js

```diff
@@ -0,0 +1,81 @@
+/* eslint-disable no-underscore-dangle */
+
+function handleErrors(ctx, error) {
+  console.error(error);
+  if (error.name === 'ValidationError') {
+    ctx.badRequest('Invalid document');
+  } else if (error.name === 'CastError') {
+    // CastError was thrown since the given ID is not a valid ObjectId
+    ctx.badRequest(`Invalid id ${ctx.params.id}`);
+  } else {
+    ctx.internalServerError();
+  }
+}
+
+function handleDocResponse(ctx, doc) {
+  if (doc === null) {
+    ctx.notFound(`Document with id ${ctx.params.id} not found`);
+  } else {
+    ctx.ok(doc);
+  }
+}
+
+async function list(ctx, Model) {
+  // TODO pagination
+  try {
+    const docs = await Model.find({});
+    ctx.ok(docs);
+  } catch (error) {
+    handleErrors(ctx, error);
+  }
+}
+
+async function getOne(ctx, Model) {
+  try {
+    const doc = await Model.findById(ctx.params.id);
+    handleDocResponse(ctx, doc);
+  } catch (error) {
+    handleErrors(ctx, error);
+  }
+}
+
+async function create(ctx, Model) {
+  const requestedDoc = new Model(ctx.request.body);
+
+  try {
+    const savedDoc = await requestedDoc.save({
+      validateBeforeSave: true,
+    });
+    ctx.ok(savedDoc);
+  } catch (error) {
+    handleErrors(ctx, error);
+  }
+}
+
+async function update(ctx, Model) {
+  try {
+    const updatedDoc = await Model.findByIdAndUpdate(ctx.params.id,
```

Comment:
Didn't we agree to use 'findOneAndUpdate'?
According to their docs, they say we should use this rather than findById.

Sames to remove function

Reply:
Just as the message I told you, they both are OK (and `findById` actually uses `findOneAndUpdate`).
The problem is with `findAnyModify` which they use, and its use can be eliminated by running `mongoose.set('useFindAndModify', false);`, as mentioned [here](https://mongoosejs.com/docs/deprecations.html), which is what I've done.